internal/providers/github/github.go

```diff
@@ -115,3 +126,78 @@ func ParseV1Config(rawCfg json.RawMessage) (*minderv1.GitHubProviderConfig, erro
 
 	return w.GitHub, nil
 }
+
+func (c *RestClient) waitIfRateLimited(ctx context.Context) error {
+	blockedChan, ok := c.tokenStatStore.GetBlockingChan(c.token, db.ProviderTypeGithub)
+	if !ok {
+		return nil
+	}
+
+	select {
+	case <-blockedChan:
+		return nil
+	case <-ctx.Done():
+		return ctx.Err()
+	case <-time.After(MaxRateLimitWait):
+		return fmt.Errorf("rate limit wait exceeded for %s", c.GetOwner())
+	}
+}
+
+func (c *RestClient) checkRateLimitResponse(ctx context.Context, resp *http.Response) error {
+	if resp.StatusCode != http.StatusForbidden && resp.StatusCode != http.StatusTooManyRequests {
+		return nil
+	}
+
+	logger := zerolog.Ctx(ctx)
```

Comment:
could we extract the URL we tried to talk to to get a sense of what call was rate-limited and add that as a `Str` for the logger?